src/cfnlint/core.py

```diff
@@ -157,8 +170,8 @@ def run_checks(filename, template, rules, regions):
         try:
             matches.extend(runner.run())
         except Exception as err:  # pylint: disable=W0703
-            LOGGER.error('Tried to process rules on file %s but got an error: %s', filename, str(err))
-            exit(1)
+            msg = 'Tried to process rules on file %s but got an error: %s' % (filename, str(err))
+            UnexpectedRuleException(msg, 1)
```

Comment:
Exception should probably be raised:
https://lgtm.com/projects/g/aws-cloudformation/cfn-lint/